contracts/token/ERC7984/extensions/ERC7984Freezable.sol

```diff
@@ -41,25 +41,56 @@ abstract contract ERC7984Freezable is ERC7984 {
 
     /// @dev Internal function to freeze a confidential amount of tokens for an account.
     function _setConfidentialFrozen(address account, euint64 encryptedAmount) internal virtual {
-        _checkFreezer();
         FHE.allowThis(encryptedAmount);
         FHE.allow(encryptedAmount, account);
         _frozenBalances[account] = encryptedAmount;
         emit TokensFrozen(account, encryptedAmount);
     }
 
-    /// @dev Unimplemented function that must revert if `msg.sender` is not authorized as a freezer.
-    function _checkFreezer() internal virtual;
+    /**
+     * @dev Internal function which gets unfrozen available of the `from` account before a transfer.
+     * Working with {_update} function.
+     */
+    function _getUnfrozenAvailableFrom(address account, euint64 encryptedAmount) internal virtual returns (euint64) {
+        if (account != address(0)) {
+            return
+                FHE.select(FHE.le(encryptedAmount, confidentialAvailable(account)), encryptedAmount, FHE.asEuint64(0));
+        }
+        return encryptedAmount;
+    }
```

Comment:
what is the point in this function? Why not just override `confidentialAvailable` when necessary?

Reply:
It is different from the confidential available function. It acts as a filter. It enables clients to override that extension (eg: temporarily skip freezable extension) seamlessly:
https://github.com/OpenZeppelin/openzeppelin-confidential-contracts/blob/ed2b6fe16a28a1db49d9b573ce4cf858f8455408/contracts/token/ERC7984/extensions/ERC7984Rwa.sol#L199-L210
Also see [Improve pre update wording in freezable](https://github.com/OpenZeppelin/openzeppelin-confidential-contracts/pull/160/commits/ed2b6fe16a28a1db49d9b573ce4cf858f8455408).